### ISSUE TITLE
Gas reporter crashes after deleting or disabling previously migrated contracts

### ISSUE THREAD

**Issue by cgewecke** [cgewecke/eth-gas-reporter#194]
As reported by @fodisi in [issue comment 104][1], if there are "stale" contracts in the Truffle `build` folder - e.g. contracts which you were previously migrating but have removed - the reporter will try to fetch deployment data for them using a transaction hash that no longer exists. 

A simple work-around is to `rm -rf build`. 

Gas reporter should not assume transaction hashes in the build folder are valid. 

[1]: https://github.com/cgewecke/eth-gas-reporter/issues/104#issuecomment-591794050

**Comment by fodisi** [comment]
Hi @cgewecke 
Thank you for opening this as a separate issue. If you're OK with it, I'd love to work on the fix and contribute back to the repo.

Regarding your comment in issue #104 regarding these piece of code:

https://github.com/cgewecke/eth-gas-reporter/blob/e48894857888f7cfe1c1886bdef0f6a7916be491/lib/gasData.js#L61-L68

Do you think it would be better to use your proposed `try/catch` block around these piece of code or simply wrap line 67 around a `if`? Knowing better about the project, do you see any implications in one or another approach?

Code with the `if` wrapping line 67.
```js
          if (contract.deployed && contract.deployed.transactionHash) {
            this.trackNameByAddress(name, contract.deployed.address);
            const receipt = this.sync.getTransactionReceipt(
              contract.deployed.transactionHash
            );
            if (receipt) {
              contractInfo.gasData.push(utils.gas(receipt.gasUsed));
            }
          }
```

**Comment by cgewecke** [comment]
@fodisi Ah yes - that's the right way 🙂. 

Can we also move `this.trackNameByAddress` inside the `if`?

**Comment by fodisi** [comment]
@cgewecke 
Yes, I think it's a good idea to do so. I'll work on it later today and submit a PR with the fix.
Thanks for your guidance!

**Comment by fodisi** [comment]
PR #195 submitted to address this issue.

**Comment by cgewecke** [comment]
Published with `0.2.16`
